src/TextField/TextField.jsx

```diff
@@ -306,7 +306,17 @@ const TextField = React.createClass({
   },
 
   componentWillMount() {
-    this._uniqueId = UniqueId.generate();
+    const {
+      hintText,
+      floatingLabelText,
+      id,
+    } = this.props;
+
+    warning(hintText || floatingLabelText || id, `We don't have enough information
```

Comment:
It's safer to use two source of uniquenest. I'm taking a side by saying a random isn't enought. But that's probably enought in most cases.